build.sbt

```diff
@@ -187,6 +194,53 @@ def crossSparkSettings(): Seq[Setting[_]] = getSparkVersion() match {
   )
 }
 
+lazy val connectCommon = (project in file("spark-connect/common"))
+  .settings(
+    name := "delta-connect-common",
+    commonSettings,
+    crossSparkSettings(),
+    releaseSettings,
+    libraryDependencies ++= Seq(
+      "io.grpc" % "protoc-gen-grpc-java" % grpcVersion asProtocPlugin(),
+      "io.grpc" % "grpc-protobuf" % grpcVersion,
+      "io.grpc" % "grpc-stub" % grpcVersion,
+      "com.google.protobuf" % "protobuf-java" % protoVersion % "protobuf",
+      "javax.annotation" % "javax.annotation-api" % "1.3.2",
+
+      "org.apache.spark" %% "spark-connect-common" % sparkVersion.value % "provided",
+    ),
+    PB.protocVersion := protoVersion,
+    Compile / PB.targets := Seq(
+      PB.gens.java -> (Compile / sourceManaged).value,
+      PB.gens.plugin("grpc-java") -> (Compile / sourceManaged).value
+    ),
+  )
+
+lazy val connectServer = (project in file("spark-connect/server"))
+  .dependsOn(connectCommon % "compile->compile;test->test;provided->provided")
+  .dependsOn(spark % "compile->compile;test->test;provided->provided")
+  .settings(
+    name := "delta-connect-server",
+    commonSettings,
+    releaseSettings,
+    crossSparkSettings(),
+    libraryDependencies ++= Seq(
+      "com.google.protobuf" % "protobuf-java" % protoVersion % "protobuf",
+
+      "org.apache.spark" %% "spark-hive" % sparkVersion.value % "provided",
```

Comment:
@andreaschat-db 

If we specify the `sparkVersion` to be master (`build/sbt -DsparkVersion=master`) then it will only compile files with prefix `src/main/scala-spark-master` (https://github.com/delta-io/delta/blob/master/build.sbt#L163).

I'm not sure if there is a way to throw an error if we compile Spark Connect by mistake with an older Spark Version, I guess most softwares if we compile with the wrong jdk or python if we compile with the wrong dependencies then it will not tell us explicit but just simply fail with an error log.

Reply:
@andreaschat-db 

Thanks to @scottsand-db on `build.sbt` we managed to make the project only compiles, tests and publishes only against Spark Master now, and it does nothing for Spark Latest (other than saying that it is skipping for Delta Connect projects).

I ran the `build.sbt -DsparkVersion={latest, master} {connectServer, connectCommon}/{compile, test, publish}` and `build.sbt {compile/test/publish}` manually to verify, but I'm not sure if there is a way to automatically test it for Github Workflows

Reply:
Awesome thank you.